nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/scroll/ScrollServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     bdelbosc
+ */
+
+package org.nuxeo.ecm.core.scroll;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.nuxeo.ecm.core.api.scroll.Scroll;
+import org.nuxeo.ecm.core.api.scroll.ScrollRequest;
+import org.nuxeo.ecm.core.api.scroll.ScrollService;
+
+/**
+ * @since 11.1
+ */
+public class ScrollServiceImpl implements ScrollService {
+
+    protected final Map<String, ScrollDescriptor> descriptors;
```

Comment:
i am wondering if it's better or not (depending if we can be in a race condition) to replace it by `ConcurrentHashMap`

Reply:
The map is initialized once during the creation of the service there are no concurrency to handle.